From 96289549b3d528f2fbc91b49d48865c7b41ed710 Mon Sep 17 00:00:00 2001
From: Radu-Andrei BULIE <radu.bulie@freescale.com>
Date: Wed, 31 Oct 2012 22:38:12 +0000
Subject: [PATCH 365/518] offload_use_case: Avoid attempt to use memory after
 free

An access was tried on a pointer after it was freed. The patch fixes this issue.

Signed-off-by: Radu-Andrei BULIE <radu.bulie@freescale.com>
[Grabbed from the branch, LINUX_IR5.2.0, of
https://git.freescale.com/git-private/cgit.cgi/ppc/alu-b4860/linux.git.]
Signed-off-by: Tiejun Chen <tiejun.chen@windriver.com>
---
 drivers/staging/fsl_dpa_offload/usecases/xfrm_km.c |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/fsl_dpa_offload/usecases/xfrm_km.c b/drivers/staging/fsl_dpa_offload/usecases/xfrm_km.c
index e78e5a9..37800c2 100644
--- a/drivers/staging/fsl_dpa_offload/usecases/xfrm_km.c
+++ b/drivers/staging/fsl_dpa_offload/usecases/xfrm_km.c
@@ -316,8 +316,8 @@ static int xfrm_km_state_notify(struct xfrm_state *x, const struct km_event *c)
 					psa->sa_id, tbl_desc);
 
 			hlist_del(&psa->h);
-			kfree(psa);
 			dbgfs_remove_entry(psa->sa_id);
+			kfree(psa);
 			/* remove key from outb post enc exact match table*/
 
 		}
@@ -328,8 +328,8 @@ static int xfrm_km_state_notify(struct xfrm_state *x, const struct km_event *c)
 				pr_info("%s(%d): error removing in SA (%d)\n",
 					__func__, __LINE__, ret);
 			hlist_del(&psa->h);
-			kfree(psa);
 			dbgfs_remove_entry(psa->sa_id);
+			kfree(psa);
 		}
 		break;
 
-- 
1.7.5.4

